src/commands/api-mesh/__tests__/log-get-bult.test.js

```diff
@@ -0,0 +1,226 @@
+const { getMeshId, getPresignedUrls } = require('../../../lib/devConsole');
+
+jest.mock('fs');
+jest.mock('axios');
+jest.mock('../../../helpers', () => ({
+	initSdk: jest.fn().mockResolvedValue({}),
+	initRequestId: jest.fn().mockResolvedValue({}),
+	promptConfirm: jest.fn().mockResolvedValue(true),
+}));
+jest.mock('../../../lib/devConsole');
+jest.mock('../../../classes/logger');
+// const mockIgnoreCacheFlag = Promise.resolve(true);
+
+describe('GetBulkLogCommand', () => {
+	let parseSpy;
+
+	beforeEach(() => {
+		// Setup spies and mock functions
+		parseSpy = jest.spyOn(GetBulkLogCommand.prototype, 'parse').mockResolvedValue({
+			flags: {
+				startTime: '2024-08-29T12:00:00Z',
+				endTime: '2024-08-29T12:30:00Z',
+				filename: 'test.csv',
+				ignoreCache: false,
+			},
+		});
+
+		// initRequestId.mockResolvedValue();
+		initSdk.mockResolvedValue({
+			imsOrgId: 'orgId',
+			imsOrgCode: 'orgCode',
+			projectId: 'projectId',
+			workspaceId: 'workspaceId',
+			workspaceName: 'workspaceName',
+		});
+		getMeshId.mockResolvedValue('meshId');
+		getPresignedUrls.mockResolvedValue({
+			presignedUrls: [{ key: 'log1.csv', url: 'http://example.com/someHash' }],
+			totalSize: 2048,
+		});
+		promptConfirm.mockResolvedValue(true);
+		global.requestId = 'dummy_request_id';
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	test('throws an error if the time difference between startTime and endTime is greater than 30 minutes', async () => {
+		// Mock the file system checks even if they are not the focus of this test
+		fs.existsSync.mockReturnValue(true); // Assume the file exists
+		fs.statSync.mockReturnValue({ size: 0 }); // Assume the file is empty
+
+		// Get the current date and time
+		const now = new Date();
+
+		// Create dynamic startTime and endTime
+		const startTime = new Date(now);
+		const endTime = new Date(now);
+		endTime.setMinutes(startTime.getMinutes() + 45); // Set endTime to 45 minutes after startTime
+
+		const formattedStartTime = startTime.toISOString().slice(0, 19) + 'Z';
+		const formattedEndTime = endTime.toISOString().slice(0, 19) + 'Z';
+
+		parseSpy.mockResolvedValueOnce({
+			flags: {
+				startTime: formattedStartTime,
+				endTime: formattedEndTime,
+				filename: 'test.csv',
+				ignoreCache: false,
+			},
+		});
+
+		const command = new GetBulkLogCommand([], {});
+		await expect(command.run()).rejects.toThrow(
+			'Max duration between startTime and endTime should be 30 minutes. Current duration is 0 hours 45 minutes and 0 seconds.',
+		);
+	});
+
+	test('throws an error if startTime format is invalid', async () => {
+		parseSpy.mockResolvedValueOnce({
+			flags: {
+				startTime: '20240809123456',
+				endTime: '2024-08-29T12:30:00Z',
+				filename: 'test.csv',
+			},
+		});
+
+		const command = new GetBulkLogCommand([], {});
+
+		// Assuming your suggestCorrectedDateFormat function corrects the format to "2024-08-09T09:08:33Z"
```

Comment:
Do you have a test to verify the output of the suggestCorrectedDateFormat (e.g. we give it an invalid datetime format and it outputs the expected corrected format)? What are some incorrect values this function could correct?

Reply:
ok. let me add a test to verify all the invalid/incorrect date and time format